src/components/dialogs/notificationDialogUtils/NotificationCollaborationGrantAccessButton.jsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { useIntl } from 'react-intl';
+import { useQueryClient } from 'react-query';
+import { get } from 'lodash-es';
+import DialogContent from '@material-ui/core/DialogContent';
+
+import Button from '../../Button';
+import CustomAlert from '../../Alert';
+import usePatchCollaboration from '../../../models/collaboration/usePatchCollaboration';
+import queryKeys from '../../../constants/queryKeys';
+// import Text from '../../Text';
+
+export default function NotificationCollaborationGrantAccessButton({
+  notification,
+  onCloseDialog,
+  path,
+}) {
+  const intl = useIntl();
+  const queryClient = useQueryClient();
+  const {
+    patchCollaborationsAsync,
+    loading,
+    error,
+    isError,
+  } = usePatchCollaboration();
+  const collaborationId = get(notification, [
+    'message_values',
+    'collaboration_guid',
+  ]);
+  return (
+    // <Text>Hi there</Text>
+    <DialogContent>
+      {/* I know that this is a nested DialogContent. If you have better advice for how to handle the fact that isError now lives locally withing these button files instead of the in the NotificationDetailsDialog, I'm all ears */}
+      <Button
+        loading={loading}
+        display="primary"
+        onClick={async () => {
+          const response = await patchCollaborationsAsync(
+            collaborationId,
+            [
+              {
+                op: 'replace',
+                path,
+                value: 'approved',
+              },
+            ],
+          );
+          if (response?.status === 200) {
+            onCloseDialog();
+            queryClient.invalidateQueries(queryKeys.me);
+          }
+        }}
+        id="GRANT_ACCESS"
+      />
+      {isError && (
+        <CustomAlert
```

Comment:
i feel like this error will show up down by the buttons? that seems like a weird place for it, though i could be wrong

Reply:
I tried it the other way and ended up moving the error here. I wish that I remembered why. I'll try it in the parent dialog file and see whether it works now.